src/target/riscv/riscv.c

```diff
@@ -3071,20 +3071,38 @@ static int riscv_read_phys_memory(struct target *target, target_addr_t phys_addr
 static int riscv_read_memory(struct target *target, target_addr_t address,
 		uint32_t size, uint32_t count, uint8_t *buffer)
 {
+	uint32_t current_count = 0;
```

Comment:
There is no need to declare variables at the top of a function. If you like, the declaration can be moved closer to the use of the variable.

Reply:
Thanks. Addressed.